src/input/type_record.cc

```diff
@@ -76,9 +76,11 @@ bool Default::check_validity(std::shared_ptr<TypeBase> type) const
 		storage_ = reader.get_storage()->get_item(0);
 		return true;
 	} catch ( Input::ReaderToStorage::ExcNotJSONFormat &e ) {
-		THROW( ExcWrongDefaultJSON() << EI_DefaultStr( value_ ) << EI_TypeName(type->type_name()));
+		THROW( ExcWrongDefaultJSON() << EI_DefaultStr( value_ ) << EI_TypeName(type->type_name())
+				<< EI_Nested(make_exception_ei<Input::ReaderToStorage::ExcNotJSONFormat>(e)) );
 	} catch ( Input::ReaderToStorage::ExcInputError &e ) {
-		THROW( ExcWrongDefault() << EI_DefaultStr( value_ ) << EI_TypeName(type->type_name()));
+		THROW( ExcWrongDefault() << EI_DefaultStr( value_ ) << EI_TypeName(type->type_name())
+				<< EI_Nested(make_exception_ei<Input::ReaderToStorage::ExcInputError>(e)) );
```

Comment:
make_exception_ei should directly return EI_Nested.
test that template parameter need not to be specified.